hop/auth.py

```diff
@@ -66,28 +91,124 @@ def load_auth(config_file=None):
 
     # load config
     with open(config_file, "r") as f:
-        config = toml.loads(f.read())["auth"]
-
-    # translate config options
-    ssl = True
-    extra_kwargs = {}
-    try:
-        # SSL options
-        if "protocol" in config and config["protocol"] == "SASL_PLAINTEXT":
-            ssl = False
-        elif "ssl_ca_location" in config:
-            extra_kwargs["ssl_ca_location"] = config["ssl_ca_location"]
-
-        # SASL options
-        user = config["username"]
-        password = config["password"]
-
-        if "mechanism" in config:
-            mechanism = config["mechanism"].replace("-", "_")
+        auth_data = toml.loads(f.read())["auth"]
+
+    return _interpret_auth_data(auth_data)
+
+
+def _interpret_auth_data(auth_data):
+    """Configures Auth instances given data read from a configuration file.
+
+    Args:
+        auth_data: The data obtained from the top-level 'auth' TOML key
+
+    Returns:
+        A list of configured Auth instances.
+
+    Raises:
+        KeyError: A required key is missing from a credential entry.
+
+    """
+    if type(auth_data) is dict:
```

Comment:
I wonder if this may be better if you were to check if `auth_data` it was dict-like versus checking if the type is explicitly a dict, e.g.

```
from collections import Mapping

...

if isinstance(auth_data, Mapping):
```

Reply:
That is what I originally wanted, but I only knew how to do it through the C interface, and gave up after a while of unsuccessful searches through the docs. In retrospect I shoud have taken a hint from the C functions all having 'Mapping' in their names. Thanks!